formwidgets/MLMediaFinder.php

```diff
@@ -0,0 +1,94 @@
+<?php namespace RainLab\Translate\FormWidgets;
+
+use Backend\FormWidgets\MediaFinder;
+use RainLab\Translate\Models\Locale;
+use System\Classes\MediaLibrary;
+
+/**
+ * MLMediaFinder Form Widget
+ */
+class MLMediaFinder extends MediaFinder
+{
+    use \RainLab\Translate\Traits\MLControl;
+
+    /**
+     * @inheritDoc
+     */
+    protected $defaultAlias = 'mlmediafinder';
+    public $originalAssetPath;
+    public $originalViewPath;
+    private $mediaPath;
+
+    /**
+     * @inheritDoc
+     */
+    public function init()
+    {
+        parent::init();
+        $this->initLocale();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function render()
+    {
+        $this->actAsParent();
+        $parentContent = parent::render();
+        $this->actAsParent(false);
+
+        if (!$this->isAvailable) {
+            return $parentContent;
+        }
+
+        $this->vars['mediafinder'] = $parentContent;
+        return $this->makePartial('mlmediafinder');
+    }
+
+    /**
+     * Prepares the form widget view data
+     */
+    public function prepareVars()
+    {
+        parent::prepareVars();
+        $this->prepareLocaleVars();
+        // make root path of media files accessible
+        $this->vars['mediaPath'] = $this->mediaPath = MediaLibrary::url('/');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function loadAssets()
+    {
+        // we only load the css from parent, unfortunately the js must be reimplemented due to missing events.
```

Comment:
I don't want to merge this with copied JS, propose a PR to the October mediafinder JS to add any events that would be required to accomplish your goal here.

Reply:
Hey, @LukeTowers we changed the code, it is modular now and can be merged. 
It needs the change in the mediafinder https://github.com/octobercms/october/pull/3446 in order to work.